src/main/java/seedu/duke/command/ListNoteCommand.java

```diff
@@ -85,61 +86,80 @@ public ListNoteCommand(Boolean isAscendingOrder, ArrayList<String> tags) {
+                            .append(notebook.getNotes().get(i).getTitle())
+                            .append(InterfaceManager.LS);
                 }
             } else {
-                noteString = getSortedString(noteString, sortedNotes);
+                noteString = new StringBuilder(getSortedString(noteString.toString(), sortedNotes));
             }
-        } else {
-            Map<Tag, ArrayList<Note>> tag = tagManager.getTagMap();
-
-            // Based on user inputted tags, will store the respective values in an ArrayList
-            // E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
-            //      1 for the values corresponding to CS2113 and the other for important tag
-            List<ArrayList<Note>> values = tags.stream()
-                    .map(tag::get)
-                    .collect(Collectors.toList());
-
-            for (int i = 0; i < values.size(); i++) {
-                for (int j = 0; j < values.get(i).size(); j++) {
-                    // Account for duplicates.
-                    // In case an item has both CS2113 and Important tag
-                    if (!notes.contains(values.get(i).get(j))) {
-                        notes.add(values.get(i).get(j));
-                    }
+
+            return noteString.toString();
+        }
+
+        // Obtaining ArrayList<String> of tags and parsing it to get an ArrayList<Tag> of tags
+        Map<Tag, ArrayList<Note>> tagMap = tagManager.getTagMap();
+        ArrayList<Tag> tagList = new ArrayList<>();
+
+        for (String tag : tags) {
+            Tag currentTag = tagManager.getTag(tag);
+
+            if (tag != null) {
+                tagList.add(currentTag);
+            }
+        }
+
+        // Based on user inputted tags, will store the respective values in an ArrayList
+        // E.g. if user input 2 tags, CS2113 and important, will have 2 ArrayList
+        //      1 for the values corresponding to CS2113 and the other for important tag
+        List<ArrayList<Note>> values = tagList.stream()
+                .map(tagMap::get)
+                .collect(Collectors.toList());
+
+        for (ArrayList<Note> value : values) {
+            for (Note note : value) {
+                // Account for duplicates.
+                // In case an item has both CS2113 and Important tag
+                if (!notes.contains(note)) {
+                    notes.add(note);
                 }
             }
+        }
 
-            // Sort the tagged notes
-            ArrayList<Note> sortedTaggedNotes = (ArrayList<Note>) notes.stream()
-                    .filter((s) -> s instanceof Note)
-                    .sorted((a, b) -> a.getTitle().toLowerCase().compareTo(b.getTitle().toLowerCase()))
-                    .collect(Collectors.toList());
+        // Checking for empty notes List
+        if (notes.isEmpty()) {
+            return String.valueOf(noteString.append(COMMAND_MESSAGE_UNSUCCESSFUL));
```

Comment:
Is there a need to append to the noteString or can you just return the unsuccessful message?

Reply:
Could've just returned the string actually